fix(storefront): BCTHEME-459 fix product quantity change error

bc-alexsaiannyi opened the pull request:
#### What?

This PR addresses an issue that appears if product quantity value is incorrect (not digital value). In this case after User clicks on Increase/Decrease button NaN is shown in the field. The problem has been fixed by applying nod validation plugin for this field.

#### Tickets / Documentation
- [BCTHEME-459](https://jira.bigcommerce.com/browse/BCTHEME-459)

#### Screenshots (if appropriate)
Sorry for bad  video quality)

https://user-images.githubusercontent.com/67792608/117249352-18187100-ae4a-11eb-94ee-80fa1d50480f.mov

with Screen Reader:

https://user-images.githubusercontent.com/67792608/117249375-1d75bb80-ae4a-11eb-9b9d-32042d5f124b.mov

BC-tymurbiedukhin commented on assets/js/theme/common/product-details.js:
```diff
@@ -76,6 +93,19 @@ export default class ProductDetails extends ProductDetailsBase {
 
+    registerAddToCartValidation() {
+        this.addToCartValidator.add([{
+            selector: '[data-quantity-change] > input[id^="qty"]',
```
Proposal: use data attribute instead of id regexp here?

bc-alexsaiannyi replied:
I switched to `.form-input--incrementTotal` since noticed that it's used further in code.

BC-tymurbiedukhin commented on assets/js/theme/common/product-details.js:
```diff
@@ -287,8 +317,7 @@ export default class ProductDetails extends ProductDetailsBase {
-
+            let qty = forms.areNumbersOnly($input.val()) ? parseInt($input.val(), 10) : quantityMin;
             // If action is incrementing
             if ($target.data('action') === 'inc') {
```
Important: put this validation to forms object as well

bc-alexsaiannyi replied:
Do you mean `If action is incrementing` block, right?

bc-alexsaiannyi replied:
done

BC-tymurbiedukhin commented on assets/js/theme/common/product-details.js:
```diff
@@ -287,35 +317,20 @@ export default class ProductDetails extends ProductDetailsBase {
-                } else {
-                    qty--;
-                }
+                qty = forms.validateDecreaseAgainstMaxBoundary(qty, quantityMin);
```
validateDecreaseAgainst\Min/Boundary

bc-alexsaiannyi replied:
Thanks for pointing this out. changed)

BC-tymurbiedukhin reviewed (approved):
LGTM

yurytut1993 commented on assets/js/theme/common/models/forms.js:
```diff
@@ -22,6 +22,45 @@ const forms = {
+     * @returns {boolean}
+     *
+     */
+    areNumbersOnly(value) {
+        const re = /^\d+$/;
+        return re.test(value);
+    },
```
I believe it should better name for example as `checkAreNumbersOnly`

bc-alexsaiannyi replied:
it's been made with analogy to `notEmpty` method above. I assume we can change to `numbersOnly` if you prefer more.

yurytut1993 replied:
`numbersOnly` sounds good

bc-alexsaiannyi commented on assets/js/theme/common/models/forms.js:
```diff
@@ -22,6 +22,45 @@ const forms = {
+     *
+     */
+    validateIncreaseAgainstMaxBoundary(value, max) {
+        const raise = value + 1;
```
I think it's better to keep a noun as name for variable. incremented || incremented is a verb or adjective.

bc-alexsaiannyi commented:
Note: while defining `addToCartValidator` in assets/js/theme/common/product-details.js, line30, I added a  condition `disableSubmit: true`. But in this case the form has been validated only  through quantity field. I removed it, it makes no sense for now.
